### PR TITLE
Show instance name in output of pulse drawer

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
Follow up of #3790 to show schedule name in the title of pulse visualizer output figure.

### Details and comments
- add title both for the pulse and schedule drawer
- position of the title is calculated from bbox
- update reference figure in unittest
- add new configuration to style sheets
- add reno

### CODE REVIEW

**Comment by lcapelluto** [comment]
LGTM!

File: qiskit/visualization/pulse/matplotlib.py [hunk 1]
```diff
@@ -305,6 +305,14 @@ def draw(self, pulse: SamplePulse,
             v_max = max(max(np.abs(re)), max(np.abs(im)))
             ax.set_ylim(-1.2 * v_max, 1.2 * v_max)
 
+        bbox = ax.get_position()
+
+        if self.style.title_font_size > 0:
+            figure.suptitle('%s' % pulse.name,
```
Comment (1ucian0) [inline qiskit/visualization/pulse/matplotlib.py:311]: why not 
```suggestion
            figure.suptitle(pulse.name,
```
or 
```suggestion
            figure.suptitle(str(pulse.name),
```

**Comment by nkanazawa1989** [comment]
@1ucian0 thanks for your suggestions. I updated that part.

**Comment by lcapelluto** [comment]
Closed https://github.com/Qiskit/qiskit-terra/issues/3752

**Pull request merged**